### PR TITLE
Add default token provider to aws-amplify

### PR DESCRIPTION
**Pull request by elorzafe**
#### Description of changes

- Add default provider token provider to umbrella package
- Add fetchAuthSession API to `aws-amplify/auth`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/providers/cognito/tokenProvider/types.ts [hunk 1]
```diff
@@ -36,15 +37,15 @@ export interface AuthTokenStore {
 export interface AuthTokenOrchestrator {
 	setTokenRefresher(tokenRefresher: TokenRefresher): void;
 	setAuthTokenStore(tokenStore: AuthTokenStore): void;
-	getTokens: ({
-		options,
-	}: {
-		options?: FetchAuthSessionOptions;
-	}) => Promise<AuthTokens>;
+	getTokens: (options?: FetchAuthSessionOptions) => Promise<AuthTokens>;
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/tokenProvider/types.ts:40]:
```suggestion
	getTokens: (options?: FetchAuthSessionOptions) => Promise<AuthTokens | null>;
```

File: packages/aws-amplify/src/initSingleton.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { AmplifyV6 } from '@aws-amplify/core';
+import { LibraryOptions, ResourcesConfig } from '@aws-amplify/core';
+import { CognitoUserPoolsTokenProvider } from './auth';
+import { LocalStorage } from '@aws-amplify/core'; // TODO(v6): use Platform storage supported
+
+export const InitializedAmplifyV6 = {
+	configure(
+		resourceConfig: ResourcesConfig,
+		libraryOptions: LibraryOptions | undefined
```
Comment (jimblanc) [inline packages/aws-amplify/src/initSingleton.ts:9]:
```suggestion
		libraryOptions?: LibraryOptions
```

File: packages/aws-amplify/src/initSingleton.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { AmplifyV6 } from '@aws-amplify/core';
+import { LibraryOptions, ResourcesConfig } from '@aws-amplify/core';
+import { CognitoUserPoolsTokenProvider } from './auth';
+import { LocalStorage } from '@aws-amplify/core'; // TODO(v6): use Platform storage supported
+
+export const InitializedAmplifyV6 = {
```
Comment (jimblanc) [inline packages/aws-amplify/src/initSingleton.ts:6]: Nit: Maybe `ConfiguredAmplifyV6` or `DefaultAmplifyV6`, although I imagine that this is temporary and will become our default singleton export? Approach makes sense, only concern is that we'll need to make sure to keep the APIs in sync if anything changes on the singleton 🤔 Maybe we could do `typeOf AmplifyV6`?

File: packages/core/src/singleton/Auth/index.ts [hunk 4]
```diff
@@ -47,40 +47,35 @@ export class AuthClass {
...
-						{
-							authConfig: this.authConfig,
-							tokens,
-							options,
-						}
-					);
-			}
-		} catch (err) {
-			// TODO(v6): validate error depending on conditions it can proceed or throw
+		// Get tokens will throw if session cannot be refreshed (network or service error) or return null if
```
Comment (jimblanc) [inline packages/core/src/singleton/Auth/index.ts:56]:
```suggestion
		// Get tokens will throw if session cannot be refreshed (network or service error) or return null if not available
```

File: packages/core/src/singleton/Auth/index.ts [hunk 5]
```diff
@@ -47,40 +47,35 @@ export class AuthClass {
...
+		tokens = await this.authOptions.tokenProvider?.getTokens(options);
+		if (tokens) {
+			// getCredentialsAndIdentityId will throw if cannot get credentials (network or service error)
+			credentialsAndIdentityId =
+				await this.authOptions.credentialsProvider?.getCredentialsAndIdentityId(
+					{
+						authConfig: this.authConfig,
+						tokens,
+						authenticated: true,
+						forceRefresh: options.forceRefresh,
```
Comment (Samaritan1011001) [inline packages/core/src/singleton/Auth/index.ts:66]: Think we need to check if options is available here before accessing forceRefresh or else it shows runtime error when nothing is given to fetchAuthSession

File: packages/core/src/singleton/index.ts [hunk 6]
```diff
@@ -88,6 +89,19 @@ class AmplifyClass {
...
 export const AmplifyV6 = new AmplifyClass();
 
+/**
+ * Returns current session tokens and credentials
+ *
+ * @param options{FetchAuthSessionOptions} - Options for fetching session.
+ *
+ * @returns Returns a promise that will resolve with fresh authentication tokens.
+ */
+const fetchAuthSession = (options: FetchAuthSessionOptions) => {
```
Comment (jimblanc) [inline packages/core/src/singleton/index.ts:99]:
```suggestion
export const fetchAuthSession = (options: FetchAuthSessionOptions) => {
```

File: packages/core/src/singleton/index.ts [hunk 7]
```diff
@@ -88,6 +89,19 @@ class AmplifyClass {
...
+	return AmplifyV6.Auth.fetchAuthSession(options);
+};
+
+export { fetchAuthSession };
+
```
Comment (jimblanc) [inline packages/core/src/singleton/index.ts:104]:
```suggestion
```

**Review by jimblanc — COMMENTED** [review]
Looks good! Some nits & questions

File: packages/auth/src/index.ts [hunk 8]
```diff
@@ -38,3 +38,5 @@ export * from './providers/cognito';
 
 // Category specific types
 export * from './types';
+
+export { fetchAuthSession } from '@aws-amplify/core';
```
[the remark this answers is not in the document]
Comment (elorzafe) [reply packages/auth/src/index.ts:42]: it is!

File: packages/aws-amplify/src/initSingleton.ts (hunk 3 above)
Comment (elorzafe) [reply packages/aws-amplify/src/initSingleton.ts]: is exported as `AmplifyV6` on the `index`

File: packages/core/src/singleton/Auth/index.ts (hunk 5 above)
Comment (elorzafe) [reply packages/core/src/singleton/Auth/index.ts:66]: added the default value

File: packages/auth/src/index.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (jimblanc) [reply packages/auth/src/index.ts:42]: Oh misread it, disregard

**Review by Samaritan1011001 — APPROVED** [review]
lgtm

**Review by jimblanc — APPROVED** [review]

**Pull request merged**